notebooks/official/bigquery-ml/bqml-vertexai-model-registry.ipynb

```diff
@@ -0,0 +1,1465 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #1.    models = aiplatform.Model.list(filter=f"display_name={MODEL_NAME}")
</pre><p>Don't find the model this way as it can lead to race condition errors if other models are added before this cell runs.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/842/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Reply:
<p>Sorry I didn't understand! Could you please elaborate? :)</p>

Reply:
Sure.
You create a model earlier in the notebook and know you try to retrieve the model by getting all models in the GCP project and assuming it's the last one.

That assumption isn't always right and you will sometimes get the wrong model (created by someone else or another process). This will cause errors down the line.

Reply:
I see your point. I was using model's name as the filter and I was thinking model names are unique on Vertex AI. I updated the code and used `aiplatform.Model(model_name=model_name)` to get the model resource. This will return a single model, not a list. Does this solve the issue? :)